src/commands/node.mjs

```diff
@@ -2316,8 +2362,1051 @@ export class NodeCommand extends BaseCommand {
+          if (this.profileValuesFile) {
+            valuesArg += this.prepareValuesFiles(this.profileValuesFile)
+          }
+
+          await self.chartManager.upgrade(
+            config.namespace,
+            constants.FULLSTACK_DEPLOYMENT_CHART,
+            config.chartPath,
+            valuesArg,
+            config.fstChartVersion
+          )
+        }
+      },
+      {
+        title: 'Kill nodes to pick up updated configMaps',
+        task: async (ctx, task) => {
+          const config = /** @type {NodeUpdateConfigClass} **/ ctx.config
+          for (const /** @type {NetworkNodeServices} **/ service of config.serviceMap.values()) {
+            await self.k8.kubeClient.deleteNamespacedPod(service.nodePodName, config.namespace, undefined, undefined, 1)
+          }
+        }
+      },
+      {
+        title: 'Check node pods are running',
+        task:
+          async (ctx, task) => {
+            const subTasks = []
+            const config = /** @type {NodeUpdateConfigClass} **/ ctx.config
+            ctx.config.allNodeIds = ctx.config.existingNodeIds
+            // nodes
+            for (const nodeId of config.allNodeIds) {
+              subTasks.push({
+                title: `Check Node: ${chalk.yellow(nodeId)}`,
+                task: () =>
+                  self.k8.waitForPods([constants.POD_PHASE_RUNNING], [
+                    'fullstack.hedera.com/type=network-node',
+                    `fullstack.hedera.com/node-name=${nodeId}`
+                  ], 1, 60 * 15, 1000) // timeout 15 minutes
+              })
+            }
+
+            // set up the sub-tasks
+            return task.newListr(subTasks, {
+              concurrent: false, // no need to run concurrently since if one node is up, the rest should be up by then
+              rendererOptions: {
+                collapseSubtasks: false
+              }
+            })
+          }
+      },
+      {
+        title: 'Fetch platform software into network nodes',
+        task:
+          async (ctx, task) => {
+            // sleep 20 seconds
+            await sleep(20000)
+            ctx.config.allNodeIds = ctx.config.existingNodeIds
+            const config = /** @type {NodeUpdateConfigClass} **/ ctx.config
+            return self.fetchLocalOrReleasedPlatformSoftware(config.allNodeIds, config.podNames, config.releaseTag, task, config.localBuildPath)
+          }
+      },
+      {
+        title: 'Setup new network node',
+        task: async (ctx, parentTask) => {
+          const config = /** @type {NodeUpdateConfigClass} **/ ctx.config
+
+          const subTasks = []
+          for (const nodeId of config.allNodeIds) {
+            const podName = config.podNames[nodeId]
+            subTasks.push({
+              title: `Node: ${chalk.yellow(nodeId)}`,
+              task: () =>
+                self.platformInstaller.taskInstall(podName, config.stagingDir, config.allNodeIds, config.keyFormat, config.force)
```

Comment:
this will copy the keys into the correct directories.  with the current code version, you are copying in the old keys.  I'm surprised that the NodeUpdateTransaction has a new hash for the TLS public key, and a new public signing gossip PEM key, but it the system is still working while the node is still using the old key files?

Reply:
Updated